banzai/main.py

```diff
@@ -117,125 +121,119 @@ def parse_args(settings, extra_console_arguments=None,
 
     logs.set_log_level(args.log_level)
 
-    if not args.ignore_schedulability:
-        settings.FRAME_SELECTION_CRITERIA += settings.SCHEDULABLE_CRITERIA
-
-    pipeline_context = PipelineContext(args, settings, **kwargs)
+    runtime_context = Context(args)
 
-    return pipeline_context
+    return runtime_context
 
 
-def run(image_path, pipeline_context):
+def run(image_path, runtime_context):
     """
     Main driver script for banzai.
     """
-    image = read_image(image_path, pipeline_context)
-    stages_to_do = get_stages_todo(pipeline_context.ORDERED_STAGES,
-                                   last_stage=pipeline_context.LAST_STAGE[image.obstype],
-                                   extra_stages=pipeline_context.EXTRA_STAGES[image.obstype])
+    image = read_image(image_path, runtime_context)
+    stages_to_do = get_stages_todo(settings.ORDERED_STAGES,
+                                   last_stage=settings.LAST_STAGE[image.obstype],
+                                   extra_stages=settings.EXTRA_STAGES[image.obstype])
     logger.info("Starting to reduce frame", image=image)
     for stage in stages_to_do:
-        stage_to_run = stage(pipeline_context)
+        stage_to_run = stage(runtime_context)
         image = stage_to_run.run(image)
     if image is None:
         logger.error('Reduction stopped', extra_tags={'filename': image_path})
         return
-    image.write(pipeline_context)
+    image.write(runtime_context)
     logger.info("Finished reducing frame", image=image)
 
 
-def run_master_maker(image_path_list, pipeline_context, frame_type):
-    images = [read_image(image_path, pipeline_context) for image_path in image_path_list]
-    stage_to_run = pipeline_context.CALIBRATION_STACKER_STAGE[frame_type](pipeline_context)
+def run_master_maker(image_path_list, runtime_context, frame_type):
+    images = [read_image(image_path, runtime_context) for image_path in image_path_list]
+    stage_constructor = import_utils.import_attribute(settings.CALIBRATION_STACKER_STAGE[frame_type])
+    stage_to_run = stage_constructor(runtime_context)
     images = stage_to_run.run(images)
     for image in images:
-        image.write(pipeline_context)
+        image.write(runtime_context)
 
 
-def process_directory(pipeline_context, raw_path, image_types=None, log_message=''):
+def process_directory(runtime_context, raw_path, image_types=None, log_message=''):
     if len(log_message) > 0:
         logger.info(log_message, extra_tags={'raw_path': raw_path})
     image_path_list = image_utils.make_image_path_list(raw_path)
     if image_types is None:
         image_types = [None]
     images_to_reduce = []
     for image_type in image_types:
-        images_to_reduce += image_utils.select_images(image_path_list, pipeline_context, image_type)
+        images_to_reduce += image_utils.select_images(image_path_list, runtime_context.db_address, image_type)
```

Comment:
The arguments for `select_images` defined in `image_utils` are different than the ones you passed here